### ISSUE TITLE
Istio Operator does not recover properly when pod is terminated (e.g. deployed in preemptible nodes) 

### ISSUE THREAD

**Issue by jahernandezmartinez13** [istio/istio#24545]
**Bug description**
When installing and configuring Istio with the operator in a cluster with preemptible nodes the behaviour is inconsistent.
We've detected that when the new pod is created, because the old was preempted, the new one overwrites the configuration to the default values. This means that if we modify the default profile with, fore example, higher HPA values, when the operator pod dies, the new one sets Istio to the default values (no HPA, prometheus pod enabled..).
The workaround consists in setting the desired HPA to an auxiliary value and then turning back to the desired (forcing the operator to detect the discrepancies), but we'll be forced to remove the preemptible nodes.

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
When the operator pod is killed it shall be created without altering the current configuration

**Steps to reproduce the bug**
1- Install Istio with Istio operator (`istioctl operator init`)
2- Configure the custom Istio values with Higher HPA (`kubectl apply -f profile.yaml`)
3- Kill the operator (delete the operator pod)
4- The new operator pod starts modifying incorrectly the istiod and ingress values to the default ones.
5- Fix step one: Modify the HPA to an auxiliar values (e.g. HPA 2-7)
5- Fix step two: Restore the desired HPA (e.g. 3-6)

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
$ istioctl version --remote
client version: 1.6.1
control plane version: 1.6.1
data plane version: 1.6.1 (19 proxies)
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-21T14:50:54Z", GoVersion:"go1.14.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-gke.15", GitCommit:"8ce551a0c972a9f113e326df0537aa3f1798d93e", GitTreeState:"clean", BuildDate:"2020-05-05T18:38:03Z", GoVersion:"go1.12.17b4", Compiler:"gc", Platform:"linux/amd64"}

**How was Istio installed?**
istioctl operator init
kubectl apply -f profile.yaml
``` profile.yaml
---
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: content-controlplane
spec:
  profile: default
  addonComponents:
    kiali:
      enabled: false
    grafana:
      enabled: false
    prometheus:
      enabled: false
    tracing:
      enabled: false
  components:
    egressGateways:
      - name: istio-egressgateway
        enabled: false
    ingressGateways:
      - name: istio-ingressgateway
        enabled: true
        k8s:
          hpaSpec:
            maxReplicas: 6
            minReplicas: 3
          resources:
            limits:
              cpu: 2000m
              memory: 1024Mi
            requests:
              cpu: 100m
              memory: 128Mi
    pilot:
      enabled: true
      k8s:
        hpaSpec:
          maxReplicas: 6
          minReplicas: 3
        resources:
          requests:
            cpu: 500m
            memory: 2Gi
  meshConfig:
    # Enable Access logs in the proxies
    accessLogFile: /dev/stdout
  values:
    sidecarInjectorWebhook:
      # Rewrite globally pod spec to send K8s probes to Pilot, so that mTLS work
      rewriteAppHTTPProbe: true
    gateways:
      istio-ingressgateway:
        sds:
          enabled: true

```

**Environment where bug was observed (cloud vendor, OS, etc)**
Google Cloud Platform, Google Kubernetes Engine (GKE)

**Operator logs after killing it**
[istio-operator.log](https://github.com/istio/istio/files/4751239/istio-operator.log)

**Screenshots**
Pre-kill
<img width="644" alt="pre-kill" src="https://user-images.githubusercontent.com/50657683/84131048-5774ec80-aa44-11ea-83e0-536594ded125.png">
Post-kill
<img width="675" alt="post-kill" src="https://user-images.githubusercontent.com/50657683/84131523-f26dc680-aa44-11ea-854e-9a243b597b03.png">

**Comment by ostromart** [comment]
I cannot repro and it's hard to see how something so basic could be broken:

```
$ ic operator init --charts manifests --hub=docker.io/istio --tag=1.6.3
$ kc create ns istio-system
namespace/istio-system created
$ kc apply -f <your yaml file above>

$ kc get hpa -n istio-system istiod
NAME     REFERENCE           TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
istiod   Deployment/istiod   <unknown>/80%   3         6         3          56s
$ kc get hpa -n istio-system istio-ingressgateway
NAME                   REFERENCE                         TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
istio-ingressgateway   Deployment/istio-ingressgateway   <unknown>/80%   3         6         3          50s

```

**Comment by juicemia** [comment]
Just dropping a comment here because we recently ran into this issue ourselves using version 1.9.8.

I'm currently doing an RCA for an outage we had. In the outage all our services went down at the same time. When we looked at Istio we noticed the Egress Gateways had disappeared, so we checked the `IstioOperator` resource and saw that it was completely blank. We were able to fix by doing a purge using `istioctl`, then reinstalling the operator and re-running all our CI to recreate all the configuration.

Looking through our logs, I can see that the `istio-operator` container was recreated (for reasons we don't know yet) right before the outage happened. I can also see messages about a leader election going wrong, which doesn't make sense considering our `istio-operator` has only one replica deployed.

I found this issue in a Google search for "istio-operator rewriting operator manifest" and it seems to report the exact behavior we just saw.
